lib/rspec/core/formatters.rb

```diff
@@ -249,5 +251,15 @@ def underscore(camel_cased_word)
       word.downcase!
       word
     end
+
+    def file_at(path)
```

Comment:
It's a little weird for the arg to be called `path` when it could be an `OutputWrapper` (which is definitely not a path!).  Also, I don't think `file_at` is the best name anymore given in the wrapper case it doesn't return a file. 
 Finally, instead of assuming it's a string in the non-wrapper case, I'd prefer we use a `case` statement for both to make fewer unproven assumptions.  Putting those together:

``` ruby
def open_file(path_or_output_wrapper)
  case path_or_output_wrapper
  when RSpec::Core::OutputWrapper
    path_or_output_wrapper = file_at(path_or_output_wrapper.output)
    path_or_output_wrapper
  when String
    RSpec::Support::DirectoryMaker.mkdir_p(File.dirname(path_or_output_wrapper))
    File.new(path_or_output_wrapper, 'w')
  end
end
```

The one thing I'm not sure on is if we should have an `else` that raises an explicit error, or just let it be `nil`.

Thoughts?